src/DynamoCore/Graph/Nodes/NodeModel.cs

```diff
@@ -917,7 +917,12 @@ public bool DisplayLabels
         [JsonIgnore]
         public bool AreAllOutputsConnected
         {
-            get { return outPorts.All(p => p.IsConnected); }
+            get { 
+                return 
+                    outPorts != null &&
+                    outPorts.Count > 0 &&
+                    outPorts.All(p => p.IsConnected); 
```

Comment:
I think the All() checks for null and empty elements because it would return false if all the elements do not satisfy the condition. Did you notice any edge cases?

Reply:
There is an edge case where a code block is created in manual mode and this property returns true even though the outports have zero elements. It assumes that all the outports are connected, but there aren't any.
![debugCode](https://user-images.githubusercontent.com/89042471/180501619-256d8a68-ca02-430d-a96c-9ec005f18010.gif)